xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -256,6 +265,9 @@ func (b *clusterImplBalancer) UpdateClientConnState(s balancer.ClientConnState)
 	// Notify run() of this new config, in case drop and request counter need
 	// update (which means a new picker needs to be generated).
 	b.pickerUpdateCh.Put(newConfig)
+	// Wait for LB config update to be done
+	<-b.cfgUpdateDone
```

Comment:
I think what we want would be the following:

- Add a new field to the `clusterImplBalancer` and this would need to be protected with `mu`
  - `inhibitPickerUpdates`
    - This needs to be set to `true` at the top of `UpdateClientConnState` to inhibit picker updates from the child LB policy when the parent is processing an update of its own.
    - This will set to `false` once the config update processing is complete (see below for more details on this)
- In `UpdateClientConnState`, we need to do the following:
  - Push the new configuration to the child policy and if the child policy returns an error, return the same from this method.
  - Enqueue the config for asynchronous processing, via a call to `b.pickerUpdateCh.Put(newConfig)`.
  - Push another event on the same channel. This would look very similar to the `resumePickerUpdates` event in the RLS LB policy. This event would contain a `done` channel that would be closed when it is handled by the `run()` goroutine.
  - Block on the `done` channel in the newly pushed `resumePickerUpdates`
  - Once unblocked, return
- In the `run` goroutine, as part of handling `resumePickerUpdates`, we need to build and send a new picker to the channel. Once the new picker is pushed, the `done` channel needs to be closed.

Reply:
Thanks @easwars for sharing this. This would be better way to ensure the picker is updated on config update. I have updated the PR, please have a look.